src/swe/swe_petsc.c

```diff
@@ -591,15 +591,15 @@ PetscErrorCode AddSWESourceTerm(RDy rdy, Vec F) {
         tby = (hv + dt * Fsum_y - dt * bedy) * factor;
       }
 
-      f_ptr[icell * ndof + 0] += water_src_ptr[icell];
-      f_ptr[icell * ndof + 1] += -bedx - tbx;
-      f_ptr[icell * ndof + 2] += -bedy - tby;
+      f_ptr[icell * ndof + 0] += swe_src_ptr[icell * ndof + 0];
+      f_ptr[icell * ndof + 1] += -bedx - tbx + swe_src_ptr[icell * ndof + 1];
+      f_ptr[icell * ndof + 2] += -bedy - tby + swe_src_ptr[icell * ndof + 0];
```

Comment:
Should the index for `swe_src_ptr` be `icell * ndof + 2` here?

Reply:
Yes, you are correct.